Add truncation to strings that go over length limits

rocket3989 opened the pull request.

rocket3989 committed:
Add truncation to strings that go over length limits

rocket3989 committed:
add newline padding to pagination check

kzfubar commented on shame_script.py:
```diff
@@ -163,7 +163,7 @@ async def paginate_message_send(
     page_length = 0
     for line, content in enumerate(message_content):
-        if len(content) + page_length > max_page:
+        if len(content) + 1 + page_length > max_page:
```
might be good to add a comment or otherwise indicate why we need page_length + 1 here

kzfubar reviewed (dismissed):
lgtm

kzfubar reviewed (approved):
lgtm

kzfubar committed:
Truncate all messages sent to discord to 2k char

Discord will not allow messages to be sent greater than 2k chars. As a
final safety check, add a safe send method that will truncate all the
messages we send.

Ideally we should provide messages less than 2k char, but this will be a
good catch for missed cases.

rocket3989 committed:
add comment, format fix

rocket3989 committed:
final cleanup for merge
